### PR TITLE
set project purpose on project page

### LINKED ISSUE #931: Ability to set project purpose on project page
Similar to adding project to organization, project managers or anyone who has editing access should be able to set purpose for their project.

### PR DESCRIPTION
**Pull request by psh0078**
fixes #931. 

Screenshots:

image

image

image

image

### CODE REVIEW

**Comment by psh0078** [comment]
I am stuck on this error where modal becomes null when trying to be closed.

Screenshot 2024-07-17 152827

File: backend/LexBoxApi/GraphQL/ProjectMutations.cs [hunk 1]
```diff
@@ -311,6 +311,28 @@ await dbContext.ProjectUsers
...
+        [Service] ProjectService projectService,
+        LexBoxDbContext dbContext)
+    {
+        await permissionService.AssertCanManageProject(input.ProjectId);
+        var project = await dbContext.Projects.FindAsync(input.ProjectId);
+        NotFoundException.ThrowIfNull(project);
+
+        project.RetentionPolicy = input.RetentionPolicy;
+        project.UpdateUpdatedDate();
+        projectService.InvalidateProjectConfidentialityCache(input.ProjectId);
```
Comment (rmunn) [inline backend/LexBoxApi/GraphQL/ProjectMutations.cs:331]: The project confidentiality cache is not affected by the retention policy changing; this line can be removed.

File: frontend/src/routes/(authenticated)/project/create/+page.svelte [hunk 2]
```diff
@@ -198,6 +198,8 @@
       bind:value={$form.retentionPolicy}
       error={$errors.retentionPolicy}
     >
+      <!-- Unknown option in temporary use for testing -->
+      <option value={RetentionPolicy.Unknown}>{'Unknown'}</option>
```
Comment (rmunn) [inline frontend/src/routes/(authenticated)/project/create/+page.svelte:202]: This should be removed once testing is complete.

**Review by rmunn — CHANGES_REQUESTED** [review]
Couple changes needed, then this should be good to go,

**Review by rmunn — APPROVED** [review]
LGTM now. Good work.

**Pull request merged**